### ISSUE TITLE
Gimme-aws-creds no longer works in Powershell after updating from 2.3.1 to 2.3.3

### ISSUE THREAD

**Issue by privojaime** [Nike-Inc/gimme-aws-creds#188]
2.3.1 was running perfectly today in Powershell.  As soon as I ran `pip3 install --upgrade gimme-aws-creds`, I type in gimme-aws-creds and I get nothing.  Not sure how I can see the errors, or force an install of the older version (I'm currently working on that)

Thanks!
-Jaime

**Comment by bicasoft** [comment]
I got the same error. And nothing seems to fix it.

**Comment by curtis-turner** [comment]
Just adding so we can see how many are impacted by this. I have the save issues, I rolled back pip uninstall gimme-aws-creds and then pip install gimme-aws-creds==2.3.1 and that seems get things working again.

**Comment by Sector95** [comment]
In 2.3.2 there was a change to the batch file that enforced the use of Python 3 in environments where more than one version of Python is installed.

#185 

Can you guys elaborate more on your Windows environment please?

**Comment by curtis-turner** [comment]
Hey @Sector95 my environment is as follows

I only have Python 3.8.2 installed, windows 10, I manage my dependencies using pipenv.
I flipped the echo OFF to ON to see the output and here is what i got.

"C:\Users\Curtis Turner\AppData\Local\Microsoft\WindowsApps\python3.exe" -x  "C:\Users\Curtis Turner\.virtualenvs\aws_scripts-15QqIJ3N\Scripts\gimme-aws-creds.cmd"

So it looks like it is detecting a different version of python 3 than it should I am looking into how to remove that if possible. hopefully this helps let me know if you need any more details.

**Comment by Sector95** [comment]
Hm.  Looks like the space in your user folder's name is causing issues...  Let me see what I can find.

**Comment by bicasoft** [comment]
Hi Sector95, I also have Python 3.8.2 and Windows 10.
I have installed Python via Chocolatey v0.10.15

**Comment by Sector95** [comment]
Can you guys pull down this branch and test it for me?
https://github.com/Nike-Inc/gimme-aws-creds/tree/win-path-fix

`pip uninstall gimme-aws-creds`
`cd \path\to\gimme-aws-creds`
`pip install .`

**Comment by curtis-turner** [comment]
Just tested it got another error.

`
gimme-aws-creds
The syntax of the command is incorrect.
`

**Comment by Sector95** [comment]
Thanks for the testing help!

I _think_ we've got it this time, if you would uninstall, reinstall, and test like above one more time, I'd appreciate it!

**Comment by curtis-turner** [comment]
Sure thing give me a few minutes and i'll report back.

**Comment by curtis-turner** [comment]
I think I got it fixed with some messing about with the cmd file let me pull the repo and push my edits.

this seemed to make it work  -> "tokens=1 delims=^"

**Comment by Sector95** [comment]
Odd, that was working on my Windows machine...  Makes me nervous that we're getting such divergent behavior.

The odd thing with that solution is that we've opened the quotes, then escaped the closing quote...  I'll try it on my machine as well.

**Comment by curtis-turner** [comment]
Yeah that does seem weird.... I think I might move to just using the docker container for this since that should at least isolate the environment to inside the container.

**Comment by curtis-turner** [comment]
I just tested it again with this change - "tokens=1 delims=" and it seems to work as well so that might be a better solution as it does not escape the last  double quote.

**Comment by Sector95** [comment]
That behavior doesn't work on my end, because the output of the `ftype` command on my computer has multiple values output:

```C:\WINDOWS\system32>assoc .py
.py=Python.File

C:\WINDOWS\system32>ftype Python.File
Python.File="C:\WINDOWS\py.exe" "%L" %*```

`delims=` causes it to capture the whole line, whereas we just want the first object in quotes.

**Comment by curtis-turner** [comment]
I moved my Python install to my root C:\ and I get the same results from those commands for the file association. This also gets rid of the space in the path issue as its just at C:\Python38-32

assoc .py
.py=Python.File

ftype Python.File
Python.File="C:\Windows\py.exe" "%L" %*

**Comment by PrivoChapin** [comment]
Greetings- any updates on the best way to get the latest version working on Windows 10? I'm still having issues past 2.3.1. Thank you!

**Comment by kylecompassion** [comment]
Would love to see this fixed so we can reliably use >2.3.1 on Windows machines

**Comment by iniquityx2** [comment]
agreed.. I have not been able to update for a while now.  I tried again yesterday to see if the issue reported in https://github.com/Nike-Inc/gimme-aws-creds/issues/302 would be resolved by the upgrade. It wasn't of course.

Now that the proposed fix is being merged, if it is not packported to 2.3.1 then windows users will still have issues.

Of course we can manually update the code but this is not ideal.

**Comment by PrivoChapin** [comment]
> Greetings- any updates on the best way to get the latest version working on Windows 10? I'm still having issues past 2.3.1. Thank you!

Not sure of the reasonings - but between Python 3.9.6 and gimme-aws-creds 2.4.3 this is now working on Windows 10 for me, where I was previously stuck on 2.3.1.

**Comment by iniquityx2** [comment]
@PrivoChapin  I tried that fix but it didn't pan out for me.  I used the windows 64 installer on (https://www.python.org/downloads/release/python-396/), chose to add python to environment checkbox of course.  Then did pip3 install --upgrade gimme-aws-creds

However, after that changed the error and that lead me to this stack exchange article.

error:  

"Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases."

https://stackoverflow.com/questions/65348890/python-was-not-found-run-without-arguments-to-install-from-the-microsoft-store

Using the second suggestion "3.Go to -> "start" and type "Manage App Execution Aliases" .Go to it and turn off "python"

and unchecking both of the python radio dots worked.

**Comment by iniquityx2** [comment]
which is also apparently solved here lol

https://github.com/Nike-Inc/gimme-aws-creds/issues/257
